### DISCUSSION TITLE
Activation Code not found with AudibleSync

### DISCUSSION THREAD

**Discussion by Jasemat05** [audiamus/AaxAudioConverter#118] [Q&A]
Hi all,

I'm having problems starting the app together with AudibleSync.
I logged in in AudibleSync and I can see my library, all right with that.
But the app still asks me to enter my activation code when starting.
does someone knows a solution?

Greetings

**Comment by audiamus** [comment]
I am not sure how AudibleSync works but it may only download your activation code if you actually connect a compatible device. However, where it would store it I have no idea. 

There are several ways to get this code. One is https://github.com/mkb79/audible-cli, a Python command line tool. 

An alternative method is to download the book directly. That's what [Book Lib Connect](https://github.com/audiamus/BookLibConnect) does, the new companion to AAX Audio Converter.

**Reply by Jasemat05** [reply]
Thanks, I'll try it.

**Comment by Jasemat05** [comment]
I downloaded and installed book lib connector now? What do I have to do next? When I enter AAX Converter, it still don't work.

**Reply by audiamus** [reply]
Did you download your book with Book Lib Connect and export it to aax? Because that's the point. With Book Lib Connect we no longer need an activation code. AAX Audio Converter will still ask for one if you have not used it previously with the Audible Win10 app or legacy Audible Manager. But you can enter any dummy code now, like 12345678. Books successfully downloaded with Book Lib Connect will already be decrypted. 

Make sure you open the Book Lib Connect-exported book in AAX Audio Converter, not the one downloaded with AudibleSync.

**Comment by Jasemat05** [comment]
Yes, it worked! Thank you very much!
And thank you for your excellent products.

**Comment by Jasemat05** [comment]
One last question ;)
Can I somehow disable that the books are downloaded in m4b format in Book Lib Connect?

**Reply by audiamus** [reply]
No, you can't and you wouldn't really want to. :sunglasses:  

The books you download with Book Lib Connect are slightly different from the books you download via your browser or AudibleSync, in terms of the encryption key. Audible calls these files AAXC, compared to AAX you download the classic way.  However, the audio book file format is always an MP4 container, and the audio stream within is always AAC. But the AAC is encrypted when it arrives on your disk. The cipher is AES (Advanced Encryption Standard) in CBC mode (Cipher block chaining), What Book Lib Connect does after downloading the file is to invoke a decrypting function, kindly provided by [AAXClean](https://github.com/Mbucari/AAXClean), a different Open Source project, that leaves the MP4 container alone, but replaces the encrypted AAC stream with a plain audio AAC stream. (AAXClean is a C# port of the original C code for AAX/AAXC in [FFmpeg](https://www.ffmpeg.org/), one of the most versatile and powerful audio and video converters, also Open Source, of course.)      

So we have this MP4 file with plain audio as the output. People and tools typically associate MP4 with video. But ours are audio-only MP4s. So, by convention the file type is changed to make this clear. Originally, audio-only MP4 was named M4A, but later a distinction was made: M4A for music, M4B for audio books. Again, this is only a file type naming convention, the file is always MP4.

So that's why the output of Book Lib Connect is M4B, an MP4 audio book.

Any change of audio format, like converting to MP3, which is offered by AAX Audio Converter, is called a transcoding. As the final goal for Book Lib Connect is to become an integrated component of AAX Audio Converter, there is no need to add transcoding functionality to Book Lib Connect. 

And then the export for AAX Audio Converter: That's just done for convenience. The AAX file you see after the export in Book Lib Connect is exactly the same as the M4B file. But as AAX Audio Converter does not accept anything else than AAX input at the moment, Book Lib Connect copies and renames the file. (It also extracts meta data from its local database, to provide AAX Audio Converter with chapter titles and series information, the additional JSON files you see.)
